packages/vite-plugin-svelte/package.json

```diff
@@ -7,18 +7,18 @@
     "src"
   ],
   "type": "module",
-  "types": "src/index.d.ts",
+  "types": "types/index.d.ts",
   "exports": {
     ".": {
-      "types": "./src/index.d.ts",
+      "types": "./types/index.d.ts",
       "import": "./src/index.js"
     },
     "./package.json": "./package.json"
   },
   "scripts": {
     "dev": "pnpm typecheck --watch",
     "build": "pnpm typecheck",
-		"typecheck": "tsc --noEmit --allowJs --checkJs"
+    "typecheck": "node scripts/generate-dts.js"
```

Comment:
the script would have to be called something like build:types or generate:types to make it more clear that it has output. check suggests it doesn't change.

When exactly would this have to be called during release, how do you ensure that it's output in /types stays in sync with changes to types in /src

Reply:
elsewhere i'm using `prepublishOnly` for this

Reply:
that does not ensure it stays in sync. You would have to call it in `prepare`  (similar to how svelte called build in prepare). and even that isn't enough because types can still change and you end up with revisions where they are out of sync.